k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/artifactSupplier/GitArtifactSupplier.kt

```diff
@@ -0,0 +1,96 @@
+package com.amazon.spinnaker.keel.k8s.artifactSupplier
+
+import com.amazon.spinnaker.keel.k8s.GIT
+import com.amazon.spinnaker.keel.k8s.model.GitRepoArtifact
+import com.amazon.spinnaker.keel.k8s.service.IgorArtifactServiceSupplier
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.artifacts.*
+import com.netflix.spinnaker.keel.api.plugins.SupportedArtifact
+import com.netflix.spinnaker.keel.api.plugins.SupportedSortingStrategy
+import com.netflix.spinnaker.keel.api.support.EventPublisher
+import com.netflix.spinnaker.keel.artifacts.BaseArtifactSupplier
+import com.netflix.spinnaker.keel.igor.artifact.ArtifactMetadataService
+import kotlinx.coroutines.runBlocking
+
+class GitArtifactSupplier (
+    override val artifactMetadataService: ArtifactMetadataService,
+    override val eventPublisher: EventPublisher,
+    private val igorArtifactService: IgorArtifactServiceSupplier
+) : BaseArtifactSupplier<GitRepoArtifact, GitRepoVersionSortingStrategy>(artifactMetadataService) {
+
+    override val supportedArtifact = SupportedArtifact("git", GitRepoArtifact::class.java)
+    override val supportedSortingStrategy =
+        SupportedSortingStrategy("git", GitRepoVersionSortingStrategy::class.java)
+
+    override fun getArtifactByVersion(artifact: DeliveryArtifact, version: String): PublishedArtifact? {
+        return runBlocking {
+            if (artifact is GitRepoArtifact) {
+                val versionFromIgor = igorArtifactService.getGitVersion(artifact.gitType, artifact.project, artifact.repoName, version )
+                return@runBlocking versionFromIgor.toPublishedArtifact(artifact.reference)
```

Comment:
do you need `return@runBlocking`? you can probably return from the entire thing?

Reply:
`return@runBlocking` isn't needed if it was not in `if` block.